src/Multi_filtration/include/gudhi/One_critical_filtration.h

```diff
@@ -0,0 +1,1378 @@
+ * - `std::numeric_limits<One_critical_filtration<T> >::infinity()` returns @ref One_critical_filtration<T>::inf() "",
+ * - `std::numeric_limits<One_critical_filtration<T> >::minus_infinity()` returns
+ *   @ref One_critical_filtration<T>::minus_inf() "",
+ * - `std::numeric_limits<One_critical_filtration<T> >::max()` throws,
+ * - `std::numeric_limits<One_critical_filtration<T> >::max(n)` returns a @ref One_critical_filtration<T> with `n`
+ *   parameters evaluated at value `std::numeric_limits<T>::max()`,
+ * - `std::numeric_limits<One_critical_filtration<T> >::quiet_NaN()` returns @ref One_critical_filtration<T>::nan() "".
+ *
+ * One critical simplicial filtrations are filtrations such that the lifetime of each object is a positive cone, e.g.
+ *  - \f$ \{ x \in  \mathbb R^2 : x>=(1,2)\} \f$ is valid, while
+ *  - \f$ \{ x \in  \mathbb R^2 : x>=(1,2)\} \cap \{x \in \mathbb R^2 : x>=(2,1)\} \f$ is not.
+ *
+ * If the lifetime corresponds to a union of such positive cones, the filtration is called a multi-critical filtration.
+ * For those cases, use @ref Multi_critical_filtration instead.
+ *
+ * @tparam T Arithmetic type of an entry for one parameter of the filtration value. Has to be **signed** and
+ * to implement `std::isnan(T)`, `std::numeric_limits<T>::has_quiet_NaN`, `std::numeric_limits<T>::quiet_NaN()`,
+ * `std::numeric_limits<T>::has_infinity`, `std::numeric_limits<T>::infinity()` and `std::numeric_limits<T>::max()`.
+ * If `std::numeric_limits<T>::has_infinity` returns `false`, a call to `std::numeric_limits<T>::infinity()`
+ * can simply throw. Examples are the native types `double`, `float` and `int`.
+ */
+template <typename T>
+class One_critical_filtration : public std::vector<T> {
+ private:
+  using Base = std::vector<T>;
+
+ public:
+  /**
+   * @brief Type of the origin of a "lifetime cone", i.e., of a one-critical filtration value.
+   * Common with @ref Multi_critical_filtration "". In the 1-critical case, simply the class it-self.
+   *
+   * @tparam U Type of a value for one parameter within the filtration value. Default value: `T`.
+   */
+  template <typename U = T>
+  using Generator = One_critical_filtration<U>;
+
+  // CONSTRUCTORS
+
+  /**
+   * @brief Default constructor. Constructs a value at minus infinity.
+   */
+  One_critical_filtration() : Base{-T_inf} {};
+  /**
+   * @brief Constructs a vector of the size of the given number of parameters with -inf as value for each entry.
+   *
+   * @warning The vector `{-inf, -inf, ...}` with \f$ n > 1 \f$ entries is not considered as "minus infinity" (the
+   * method @ref is_minus_inf() will not return true). The `-inf` are just meant as placeholders, at least one entry
+   * should be modified by the user. Otherwise, either use the static method @ref minus_inf() or set @p n to 1 instead.
+   *
+   * @param n Number of parameters.
+   */
+  One_critical_filtration(int n) : Base(n, -T_inf) {};
```

Comment:
This (and some others below?) should probably be marked `explicit`. 
Would it cause problems to merge it with the next constructor?